xrayvision/imaging.py

```diff
@@ -296,8 +307,8 @@ def generate_header(vis: Visibility, *, shape: Quantity[apu.pix], pixel_size: Qu
     :
     """
     header = {
-        "crval1": (vis.offset[1]).to_value(apu.arcsec),
-        "crval2": (vis.offset[0]).to_value(apu.arcsec),
+        "crval1": (vis.phase_center[1]).to_value(apu.arcsec),
+        "crval2": (vis.phase_center[0]).to_value(apu.arcsec),
```

Comment:
So this is the center of the map so I want a map at 5", 5" but the phase center is 0,0 e.g. for stix vis when you create it a some location you phase shift the visibilities to be at that location so the phase centre is 0",0"?

Reply:
I think we need to discuss this in more details. Concerning this, I had to set phase_center=[0.,0.] in a couple of places ([1](https://github.com/paolomassa/xrayvision/blob/e857a0666e67db2c5c176406bc0787a029f377f9/xrayvision/imaging.py#L128) and [2](https://github.com/paolomassa/xrayvision/blob/e857a0666e67db2c5c176406bc0787a029f377f9/xrayvision/imaging.py#L170)) to pass the tests

Reply:
As a user in recent days, I've found this distinction very confusing.  I think we need documentation in the docstrings that very clearly describes and defines what each is.

Reply:
Yea I think I've mixed up phase centre and offset but I'm not sure or another concept.

Given some visibilities `V` that we create/observed with phase_center at 0,0. 

I want to make an image (not map) of 100x100" centred on 250,250". No problem I need to generate x,y coordinates from 200 to 300 (`np.meshgrid(np.arange(200, 301),np.arange(200, 301))`)and all good. But that can be a but cumbersome so a decided to shift the phase center to be at 250,250 by multiplying `V` by a phase factor `phi` to get `V'`. 

Now I want to make the same image (not map) 100x100" centred on 250,250" again no problem but now I need to generate x,y coordinates from -50 to 50 `np.meshgrid(np.arange(-50, 50),np.arange(-50, 50))` as the phase center is already at 250, 250.


So in 1) above I create a visibility that can have a non zero phase center lets say 250,250"
```
vis = dft_map(image, u=u, v=v, phase_center=[250., 250.]*apu.arcsec, pixel_size=pixel_size) 
```
The way the current code works the visibility object created should have phase_center=[0,0] so because the create vis has effectively been phase shifted as above.
```
Visibilities(vis, u=u, v=v, phase_center=[0,0])
```
not matter the given `phase_center` value give in DFT above.

In 2) I don't think it's needed as the default is 0,0? 

I'm not sure if it's not like `crval` and `crpix`

Reply:
If you want to reconstruct an image centered in [x,y] you can equivalently:
1.  have `phase_center = [x,y]` and create a meshgrid X,Y centerd in [0,0]
2. have `phase_center = [0,0]` and create a meshgrid X,Y centered in [x,y]
3. In general, have `phase_center = [x',y']` and create a meshgrid X,Y centerd in [x - x',y - y'] (or, maybe, [x' - x,y' - y]; it does not matter for this discussion)

In SSW-IDL, the standard choice is 1., so there is never confusion about `phase_center` and the center of the meshgrid. However, if we want to make the code as general as possible, we should implement 3. In this case, we would need to agree on the variable names to be used. I would leave `phase_center` as it is and introduce an `offset` variable (for instance, in `dft_map`, as suggested by @samaloney below). Then we would need to slightly adapt the code to compute the difference between `phase_center` and `offset` when we reconstruct an image.

Since so far we haven't agreed on what to do, I have considered the same strategy that is implemented in SSW-IDL, as a temporary solution.

Reply:
Yea I think ultimately option 3 is the best as it also that means that you can always get back to `V` given `V'` because you have the `offset` and `phase_center`. But we can push to a later version and make it work with option 1 for now.